commit b7c7b592e9e48146aa27316fdecd144e1555571f
Author: Alberto Villa <avilla@FreeBSD.org>
Date:   Tue Sep 18 15:46:54 2012 +0200

    Fix free space information in the device notifier
    
    The device notifier is using model["Free Space( Text)"] to show
    information which is not always available. As per Qt documentation,
    QML models need their roles defined at the beginning, thus, when the
    plasmoid is loaded with no accessible drives on the system, those two
    roles are not created
    (plasma/generic/dataengines/soliddevice/soliddeviceengine.cpp), and
    the plasmoid will always show "undefined" for its whole lifetime.
    
    REVIEW: 106490
    FIXED-IN: 4.9.2

diff --git a/plasma/generic/applets/devicenotifier/package/contents/ui/DeviceItem.qml b/plasma/generic/applets/devicenotifier/package/contents/ui/DeviceItem.qml
index 7ac95e7..e98100e 100644
--- a/plasma/generic/applets/devicenotifier/package/contents/ui/DeviceItem.qml
+++ b/plasma/generic/applets/devicenotifier/package/contents/ui/DeviceItem.qml
@@ -181,7 +181,7 @@ Item {
 
         PlasmaCore.ToolTip {
             target: freeSpaceBar
-            subText: i18nc("@info:status Free disk space", "%1 free", model["Free Space Text"])
+            subText: i18nc("@info:status Free disk space", "%1 free", sdSource.data[udi]["Free Space Text"])
         }
 
         QIconItem {
diff --git a/plasma/generic/applets/devicenotifier/package/contents/ui/devicenotifier.qml b/plasma/generic/applets/devicenotifier/package/contents/ui/devicenotifier.qml
index bf07b10..641664d 100644
--- a/plasma/generic/applets/devicenotifier/package/contents/ui/devicenotifier.qml
+++ b/plasma/generic/applets/devicenotifier/package/contents/ui/devicenotifier.qml
@@ -257,7 +257,7 @@ Item {
             state: model["State"]
 
             percentUsage: {
-                var freeSpace = new Number(model["Free Space"]);
+                var freeSpace = new Number(sdSource.data[udi]["Free Space"]);
                 var size = new Number(model["Size"]);
                 var used = size-freeSpace;
                 return used*100/size;
